### PR TITLE
feat: Fork nginx.conf on OS

### PR DESCRIPTION
**Pull request by iProgramStuff**
When running `sentry` using relay, our nginx reverse proxy points to `host.docker.internal` to talk to the host outside of the container. This works well on a Mac setup, but doesn't resolve in Linux. Docker doesn't support it.

This PR attempts to resolve this by using a different nginx conf file if we're running on Linux. This config points to `172.17.0.1` instead, which, based on some light reading, should point to the right spot. Could be wrong. But maybe this is at least a step in the right direction.

I have not tested anything. Maybe tomorrow we can ask Sarah to try this.

### CODE REVIEW

File: src/sentry/conf/server.py [hunk 1]
```diff
@@ -117,8 +118,13 @@ def env(key, default="", type=None):
...
 )
 
+if platform.system() == "Linux":
+    nginx_conf_file = "nginx.linux.conf"
+else:
+    nginx_conf_file = "nginx.conf"
+
 REVERSE_PROXY_CONFIG = os.path.normpath(
-    os.path.join(PROJECT_ROOT, os.pardir, os.pardir, "config", "reverse_proxy", "nginx.conf")
+    os.path.join(PROJECT_ROOT, os.pardir, os.pardir, "config", "reverse_proxy", "nginx_conf_file")
```
Comment (iProgramStuff) [inline src/sentry/conf/server.py]: This should be using nginx_conf_file the variable, not "nginx_conf_file" the string.

Comment (iProgramStuff) [reply src/sentry/conf/server.py]: I'll fix soon, but I'm on another branch atm and don't feel like it right now.

**Review by RaduW — CHANGES_REQUESTED** [review]
Duplicating the whole file seems a bit heavy.

I would prefer if we could place the common code (at least all the server definition) into a separate file (sever.conf) and then from the two config files (nginx.linux.conf and nginx.conf) do a 
include  /etc/nginx/server.conf

Ideally we the two config versions would only contain the different upstream definitions and include the common config (but I'm not sure that you could pull the upstream definitions  at the top of the file, we need to test that).

**Comment by iProgramStuff** [comment]
> Duplicating the whole file seems a bit heavy.
> 
> I would prefer if we could place the common code (at least all the server definition) into a separate file (sever.conf) and then from the two config files (nginx.linux.conf and nginx.conf) do a
> include /etc/nginx/server.conf
> 
> Ideally we the two config versions would only contain the different upstream definitions and include the common config (but I'm not sure that you could pull the upstream definitions at the top of the file, we need to test that).

Cool. I can definitely look into duplicating less - I agree, this was just a quick POC. Other than that, the concept seems sound, and feels like it should work to you? :-)

**Comment by joshuarli** [comment]
+1 on Duplicating the whole file seems a bit heavy.

Given that linux dev is currently a minority, it might be a better idea for now to recommend some manual `/etc/hosts` configuration.

**Comment by RaduW** [comment]
@iProgramStuff I think that the way you approached it is very good.
I have done a quick try and unfortunately you cannot place upstream definitions at the top of the config file so the easiest way I see (without creating too many files) is to have the two OS specific config files ( which will unfortunately contain some duplication) and then, inside both files, import a common file that contains the server definition part (which is the part most like to change in the future with new location definitions).

So it would be something like this:
`/etc/nginx/nginx.<os>.config`: 

```
user nginx;
...
http {
...
   upstream sentry {
     ...
   }
  ...
  include /etc/nginx/server.conf
}
```
`/etc/nginx/server.conf`:

```
server {
   listen 80;
   resolver 127.0.0.11 ipv6=off;

   location /api/store/ { ... }
   ...
}
```

**Review by RaduW — APPROVED** [review]
Nice.

**Review by untitaker — COMMENTED** [review]
I tried it in a Ubuntu VM and nginx is spewing `connection refused` while trying to connect to `"http://172.17.0.1:8888/"`.

**Comment by RaduW** [comment]
Just noticed, when @untitaker pointed up... where is 172.17.0.1 come from ?

**Comment by BYK** [comment]
This feature is coming soon and there are also a bunch of simpler workarounds here: https://github.com/docker/for-linux/issues/264#issuecomment-714253414

Closing due to that and staleness.

**Pull request closed without merging**